panacea/web/static/js/view.js

```diff
@@ -100,41 +155,122 @@ const hideDownloadButton = () => {
+  let formattedCategory = category.replace('_',' ');
+  if (formattedCategory === 'alternative') {
+    formattedCategory += ' non-DDI';
+  }
+  return formattedCategory;
+};
 
-    ddisTextElement.innerHTML = `<p>${preamble}</p><ul>${ddisHTML}</ul>`;
+const generateDdiElement = ddi => {
+  const ddiContainer = document.createElement('div');
+  let containerClass = null;
+  if (ddi.harmful) {
+    if (ddi.category === 'alternative') {
+      containerClass = 'warning';
+    } else {
+      containerClass = 'danger';
+    }
   } else {
-    ddisTextElement.innerHTML = 'I have not identified any interactions between the drugs in this file.';
+    containerClass = 'success';
   }
+  ddiContainer.classList.add('wrapper', containerClass);
+
+  const interactionContainer = Util.createElementWithClass('h5', 'ddi-title');
+  const interactionLink = document.createElement('a');
+  interactionLink.innerHTML = ddi.label;
+  interactionLink.href = ddi.uri;
+  interactionLink.target = '_blank';
+  interactionContainer.appendChild(interactionLink);
+
+  const drugsInvolvedText = `${ddi.drug_a.label} on line ${ddi.drug_a.line} and ${ddi.drug_b.label} on line ${ddi.drug_b.line}`;
+  const drugInfoElement = generateDdiInfoSnippet('Drugs Involved', drugsInvolvedText);
+  const harmfulElement = generateDdiInfoSnippet('Disposition', ddi.harmful ? 'Harmful' : 'Not Harmful');
+  const categoryElement = generateDdiInfoSnippet('Category', formatDdiCategoryAtom(ddi.category));
+  const spacingElement = generateDdiInfoSnippet('Spacing', `${ddi.spacing} days`);
+
+  ddiContainer.appendChild(interactionContainer);
+  ddiContainer.appendChild(drugInfoElement);
+  ddiContainer.appendChild(harmfulElement);
+  ddiContainer.appendChild(categoryElement);
+  ddiContainer.appendChild(spacingElement);
+
+  return ddiContainer;
+};
+
+const generateDdiInfoSnippet = (labelText, infoText) => {
+  const container = Util.createElementWithClass('div', 'ddi-info');
+  const label = Util.createElementWithClass('span', 'ddi-info-label');
+  const info = Util.createElementWithClass('span', 'ddi-info-text');
+
+  label.innerHTML = labelText;
+  info.innerHTML = infoText;
+  container.appendChild(label);
+  container.appendChild(info);
+
+  return container;
+};
+
+export const displayNoDdis = () => {
+  const ddisTextElement = document.getElementById('ddis-text');
+  ddisTextElement.innerHTML = 'I have not identified any interactions between the drugs in this file.';
   showElement(ddisPanel);
 };
 
-export const displayError = error => {
+export const displayError = (error, pml) => {
   const errorTitleElement = document.getElementById('error-title');
   const errorTextElement = document.getElementById('error-text');
+  const errorsBadge = document.getElementById('errors-badge');
+  const errorsTab = document.getElementById('errors-tab');
 
+  errorsBadge.innerHTML = '1';
+  errorsTab.click();
   errorTitleElement.innerHTML = error.title;
   errorTextElement.innerHTML = error.detail;
```

Comment:
What do you think about sticking `error.detail` in a `<strong>`?

Some other errors (no drugs) do this iirc.